strider/server.py

```diff
@@ -46,78 +49,110 @@
     "message": {
         "query_graph": {
             "nodes": {
-                "n1": {
-                    "id": "CHEBI:136043",
-                    "category": "biolink:ChemicalSubstance",
+                "n0": {
+                    "id": "MONDO:0001056",
+                    "category": "biolink:Disease"
                 },
-                "n2": {
-                    "category": "biolink:Disease",
+                "n1": {
+                    "category": "biolink:Disease"
                 }
             },
             "edges": {
-                "e12": {
-                    "subject": "n1",
-                    "object": "n2",
-                    "predicate": "biolink:treats",
+                "e01": {
+                    "subject": "n0",
+                    "object": "n1",
+                    "predicate": "biolink:related_to"
                 }
             }
-        },
+        }
     }
 }
 
-@APP.post('/query', response_model=Message, tags=['query'])
-async def sync_query(
-        query: Query = Body(..., example=EXAMPLE),
-        support: bool = True,
-) -> Message:
-    """Handle synchronous query."""
-    return await sync_answer(
-        query.dict(),
-        support=support,
+# How long we are storing results for
+store_results_for = int(os.getenv(
+    "STORE_RESULTS_FOR",
+    1 * 24 * 60 * 60,
+))
+
+
+def get_finished_query(qid: str) -> ReasonerResponse:
+    qgraph = RedisGraph(f"{qid}:qgraph")
+    kgraph = RedisGraph(f"{qid}:kgraph")
+    results = RedisList(f"{qid}:results")
+    logs = RedisList(f"{qid}:log")
+
+    qgraph.expire(store_results_for)
+    kgraph.expire(store_results_for)
+    results.expire(store_results_for)
+    logs.expire(store_results_for)
+
+    return ReasonerResponse(
```

Comment:
I've tried to avoid instantiating these pydantic models so that the internal logic is not beholden to them. FastAPI will convert dict outputs for us, so I'd just make the dict here. Of course, my strategy does not play nicely with the type hints using pydantic and pyright... so I'm not sure what the right solution is.

Reply:
Yeah that's tough because we will never be able to make pyright happy without using the models, but also there is a lot of work to get all of the internal data structures to use pydantic models. I think for now we should be consistent and use dicts everywhere, then maybe at some point move to the pydantic models.

Reply:
Sounds like a plan.